File: solutions/ruby/birthday.rb
def birthday s, d, m
	ways = 0
	sum = 0
	s.each_with_index do |el, idx|
		if (idx + m > s.size())
			break
		end
		(idx...(idx + m)).each_with_index do |curIdx|
			sum = sum + s[curIdx]
		end
		if sum == d
			ways = ways + 1
		end
		sum = 0
	end
	return ways
end

File: solutions/java/Birthday.java
public class Birthday{
	public static int run(int[] s, int d, int m){
		int ways = 0;
		int sum = 0;
		int arraySize = s.length;

		for(int i = 0; i < arraySize ; i++){
			if(i + m > arraySize) break;

			for(int j = i; j < (i + m); j++){
				sum += s[j];
			}

			if(sum == d) ways++;

			sum = 0;
		}
		return ways;
	}
}